src/mask_gen_utils/mask_inspect.py

```diff
@@ -164,75 +184,121 @@ def create_dataset(self, samples) -> fo.Dataset:
             image_name = Path(sample.filepath).name
-            if image_name in tags_map:
-                sample.tags = tags_map[image_name]
+            if image_name in tag_map:
+                sample.tags = tag_map[image_name]
 
         dataset = fo.Dataset(self.dataset_name)
         dataset.add_samples(samples)
         return dataset
 
-    def save_tags_to_db(self, output_csv: Path) -> None:
+    def _get_existing_data(self) -> dict:
+        """
+        Loads existing data from the voxel inspection results database.
+        
+        Returns:
+            dict: A dictionary mapping image names to their initial and final voxel tags.
+        """
+        existing_data = {}
+        for _, row in self.df.iterrows():
+            existing_data[row["image_name"]] = {
+                "initial_voxel_tag": row["initial_voxel_tag"],
+                "final_voxel_tag": row["final_voxel_tag"]
+            }
+        return existing_data
+
+    def _update_existing_rows(self, current_data: dict) -> None:
+        """
+        Updates self.df based on current_data if tags have changed.
+        
+        Args:
+            current_data (dict): A dictionary mapping image names to their new tags.
+        """
+        for idx, row in self.df.iterrows():
+            image_name = row["image_name"]
+            if image_name not in current_data:
+                continue
+
+            new_tags = current_data[image_name]
+            old_tags = str(row["initial_voxel_tag"]) if pd.notna(row["initial_voxel_tag"]) else ""
+
+            if old_tags == "good":
+                continue  # Do not change 'good'
+            elif new_tags == "good":
+                self.df.at[idx, "final_voxel_tag"] = new_tags
+            elif new_tags != old_tags and new_tags != "good":
+                self.df.at[idx, "initial_voxel_tag"] = new_tags
+
+    def _append_new_rows(self, current_data: dict, existing_data: dict) -> None:
+        """
+        Appends rows to self.df for any new image names not already present.
+
+        Args:
+            current_data (dict): A dictionary mapping image names to their tags.
+            existing_data (dict): A dictionary mapping existing image names to their tags.
         """
-        Updates a CSV file with image filenames and their associated tags from self.dataset.
-        Only replaces tags if they have changed; otherwise, keeps the existing tags.
-        Appends new images if not already present.
+        new_rows = []
+        for image_name, tags_str in current_data.items():
+            if image_name not in existing_data:
+                new_rows.append({
+                    "image_name": image_name,
+                    "initial_voxel_tag": tags_str,
+                    "final_voxel_tag": ""
+                })
+        if new_rows:
+            self.df = pd.concat([self.df, pd.DataFrame(new_rows)], ignore_index=True)
+
+    def _populate_empty_db(self, current_data: dict) -> None:
```

Comment:
The rows of self.df were not populating if new db needed to be created. So, I changed this methods and now it is working.